client.go

```diff
@@ -70,6 +77,14 @@ func WithHTTPClient(c *http.Client) Option {
 	}
 }
 
+// WithConnectTimeout sets the timeout when connecting to a BMC to create a new session.
```

Comment:
You would mind adding to this that this the same timeout used for each provider?